backend/src/main/java/cloudflight/integra/backend/repository/InMemoryProposalRepository.java

```diff
@@ -0,0 +1,46 @@
+package cloudflight.integra.backend.repository;
+
+import cloudflight.integra.backend.model.Proposal;
+import cloudflight.integra.backend.model.Type;
+import org.springframework.stereotype.Repository;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.UUID;
+
+@Repository
+public class InMemoryProposalRepository implements RepositoryProposalInterface{
+    private final List<Proposal> entities = new ArrayList<>();
```

Comment:
No need to change it now, but for future reference.
Changing the Data Structure Type to a Map (HashMap) here would be simpler.

We could have a `Map<String, Proposal>`, in which we keep our Proposals by `id`.
It would simplify all the other methods, because we would not need streams.

```java
return entities.stream().filter(proposal -> proposal.getId().equals(id)).findFirst();
```

becomes

```java
return entities.get(id)
```

Maps/Dictionaries are quite cool, and they are often a topic in Technical Interviews.